File: CCIP/CCIP/Class+addition/UISegmentedControl+addition.m
//
//  UISegmentedControl+addition.m
//  CCIP
//
//  Created by FrankWu on 2016/7/19.
//  Copyright © 2016年 CPRTeam. All rights reserved.
//

#import "UISegmentedControl+addition.h"

@implementation UISegmentedControl (addition)

- (void)resetAllSegments:(NSArray *)segments
{
    NSInteger oldIndex = [self selectedSegmentIndex];
    NSString *oldTitle = (oldIndex != -1) ? [self titleForSegmentAtIndex:oldIndex] : nil;
    [self removeAllSegments];
    
    for (int i = 0; i < [segments count]; ++i) {
        NSString *title = [segments objectAtIndex:i];
        [self insertSegmentWithTitle:title
                             atIndex:self.numberOfSegments
                            animated:NO];
        if ([title isEqualToString:oldTitle])
            [self setSelectedSegmentIndex:i];
    }
}

@end

File: OPass/Class+addition/UISegmentedControl+addition.swift
//
//  UISegmentedControl+addition.swift
//  OPass
//
//  Created by 腹黒い茶 on 2018/11/4.
//  2018 OPass.
//

import Foundation
import UIKit

extension UISegmentedControl {
    func resetAllSegments(_ segments: NSArray) {
        let oldIndex: NSInteger = self.selectedSegmentIndex
        let oldTitle: String? = oldIndex != -1 ? self.titleForSegment(at: oldIndex) : nil
        self.removeAllSegments()

        for i in 0..<segments.count {
            if let title: String = segments.object(at: i) as? String {
                self.insertSegment(withTitle: title, at: self.numberOfSegments, animated: false)
                if (title == oldTitle) {
                    self.selectedSegmentIndex = i
                }
            }
        }
    }
}
